crates/telemetry-subscribers/src/lib.rs

```diff
@@ -164,6 +171,11 @@ impl TelemetryConfig {
         self
     }
 
+    pub fn with_prom_registry(mut self, registry: &'reg prometheus::Registry) -> Self {
+        self.prom_registry = Some(registry);
+        self
+    }
```

Comment:
The clone of a prometheus registry is the clone of an internal Arc. It is better to clone the registry here rather than take the lifetime generic imposed on your struct by taking a reference.
https://docs.rs/prometheus/latest/src/prometheus/registry.rs.html#220